src/clims/services/substance.py

```diff
@@ -365,11 +367,25 @@ def load_file(self, organization, full_path, fileobj, add_timestamp=False):
             full_path, ext = full_path.rsplit(".", 1)
             full_path = full_path + datetime.now().strftime("%m_%d_%Y%_H_%M_%S") + "." + ext
 
-        from sentry.plugins import plugins
-        plugins.require_handler(SubstancesSubmissionHandler)
+        context = HandlerContext(organization=organization)
+        with MultiFormatFile.from_file_stream(full_path, file_stream) as wrapped_stream:
+            plugins.handle(SubstancesValidationHandler, context, False, wrapped_stream)
+
         logger = logging.getLogger('clims.files')
         logger.info('substance_batch_import.start')
 
+        org_file = self._create_organization_file(file_stream, full_path, organization.id, logger)
+
+        # Call handler synchronously and in the same DB transaction
```

Comment:
Not sure about this comment. If I understand this correctly these will not be called in the same transaction?

Reply:
This comment was here before. Probably, it looks like a new comment because I have moved some code around. Should I remove the comment?

Reply:
@withrocks This sound like a TODO thing? Or how does this actually work?